### PR TITLE
[ML] add multi:softmax|softprob XGBoost support

### PR DESCRIPTION
**Pull request by benwtrent**
This commit adds support for XGBoost multi-class classification model transformations.

XGBoost handles multiclass classification by having `n_classes * n_estimators` trees. 
Then, every `n_classes` tree corresponds to their respective classes

Since Elasticsearch supports multi-valued leaves, we can transform the xgboost format by choosing the
appropriate leaf value index given the tree id.

This commit also fixes a minor model transformation bug where if xgboost actually defined the feature_names instead of the default `f1,...fn` we would blow up. 

closes https://github.com/elastic/eland/issues/242

### CODE REVIEW

File: eland/tests/ml/test_imported_ml_model_pytest.py [hunk 1]
```diff
@@ -228,8 +228,10 @@ def test_random_forest_regressor(self, compress_model_definition):
     @pytest.mark.parametrize("compress_model_definition", [True, False])
     def test_xgb_classifier(self, compress_model_definition):
         # Train model
-        training_data = datasets.make_classification(n_features=5)
-        classifier = XGBClassifier(booster="gbtree")
+        training_data = datasets.make_classification(
```
Comment (sethmlarson) [inline eland/tests/ml/test_imported_ml_model_pytest.py:231]: Can we add an additional test (or parametrize this test) so we maintain both the binary and multi classification?

Comment (benwtrent) [reply eland/tests/ml/test_imported_ml_model_pytest.py]: @sethmlarson added a new parameter :)
